kit/src/main/java/com/oracle/javafx/scenebuilder/kit/editor/panel/library/manager/LibraryDialogController.java

```diff
@@ -147,7 +148,9 @@ void loadLibraryList() {
             listItems = FXCollections.observableArrayList();
         }
         listItems.clear();
-        libraryListView.setItems(listItems);
+        
+        SortedList<DialogListItem> sortedItems = listItems.sorted(new DialogListItemComparator());
```

Comment:
Thanks for pointing this out, never observed this. I'll rework accordingly.

Reply:
I have added a call to `libraryListView.getSelectionModel().selectFirst();` to achieve this. According to Javadoc the method will attempt to do the job, so I skipped a check for an empty collection.

Reply:
This is not working for me. I have a long list of items, and only if I move `selectFirst()` to the end, right after `listItems.addAll(...)`, then the first item is selected.

Anyway, I'm not sure we should try to select any item at all by default.

I see that we have this bug even before your applying fix, but without sorting the list, the first item is focused when the dialog is shown for the first time, and doesn't look that bad.

If you move `libraryListView.setItems(sortedItems);` to the end, the first item in the sorted list will be the one focused, so I guess this is a better option.

Reply:
I've now tried 4 variants of the setup.

Option A uses no selection after creating the list at all.

![VariantA](https://user-images.githubusercontent.com/22102800/113362900-c3b43a00-934f-11eb-9efd-63dabc6ad582.png)

Option B uses the variant where the selection is made after populating the list with values.

![VariantB](https://user-images.githubusercontent.com/22102800/113362906-c6169400-934f-11eb-89ac-1a3a4a36931e.png)

Option C additionally requests focus after populating the list with values.

![VariantC](https://user-images.githubusercontent.com/22102800/113362910-c9118480-934f-11eb-9463-641fa88e8bce.png)

The first approach where I've added the selection after creation of the SortedList is clearly not okay. It only works once, most likely by random. Afte opening/closing the window the behaviour is also not consistent. Actually my thinking was wrong here.

I agree with with your proposal and I also like the idea to have a default selection. The reason is, that you always have a defined starting point in the dialog window for keyboard navigation (except the first time when there is an empty list).
When adding a library, the selection also moves to the - possibly new - first item in the list. Here one could expect the recently added item to be selected (also the dialog looses focus after each action). Still would favor option B.

Reply:
Option B looks good. The pending issue now is that the second time you open the dialog, the listView is not focused, so that's why the first item is selected but in grey.

Adding `libraryListView.requestFocus();` at the end seems to solve it?

Reply:
Focus is now requested after loading the list.